inference-engine/thirdparty/clDNN/src/primitive_inst.cpp

```diff
@@ -161,6 +182,32 @@ primitive_inst::primitive_inst(network& network, program_node const& node, bool
     }
 }
 
+void primitive_inst::allocate_internal_buffers(void) {
+    if (_impl == nullptr) return;
+    const auto& ibuf_info = _impl->get_internal_buffer_info();
+    if (ibuf_info.sizes.empty()) return;
+
+    auto& engine = get_network().get_engine();
+    auto alloc_type = allocation_type::usm_host;
```

Comment:
I think it won't work properly if USM is not available. So maybe it should look as follows:
```cpp
auto alloc_type = engine.get_lockable_preffered_memory_allocation_type();
if (engine.supports_allocation(allocation_type::usm_device) &&
    std::accumulate(ibuf_info.sizes.begin(), ibuf_info.sizes.end(), 0) <= engine.get_device_info().max_alloc_mem_size * 0.85) {
...
}
```

Also, I'd say that we should modify `ocl_engine::allocate_memory` implementation and add a check that engine supports given allocation type.

Reply:
Okay, I'll fix.

Reply:
Fixed